app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -164,6 +169,17 @@ class BrowserToolbarIntegration(
         customTabsFeature?.start()
         navigationButtonsIntegration?.start()
         observerSecurityIndicatorChanges()
+        observeCurrentUrlChanges()
+    }
+
+    private fun observeCurrentUrlChanges() {
+        fragment.consumeFlow(store) { flow ->
+            flow.map { state -> state.selectedTab?.content?.url }
+                .ifChanged()
```

Comment:
Should this be something like:

```suggestion
            flow.map { state -> state.selectedTab?.content }
                .ifAnyChanged { arrayOf(loading, url) }
```

Or maybe just loading? We can get URL updates without navigation changes (like anchor links).

Reply:
Ah, looks like AC's [toolbar controller](https://searchfox.org/mozilla-mobile/source/android-components/components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarBehaviorController.kt#41) does just that - observes `content.loading`.

Related: I wonder why we aren't using the `TabsFeature` here and getting this for free..

Might be another one of those toolbar things to investigate after 93. 😄

Reply:
The AC handles its own toolbar, we still have in Focus a separate DisplayToolbar that currently handles the hiding on scroll...

Reply:
We can rely on `content.showToolbarAsExpanded `when we remove the Focus `DisplayToolbar` and set the behavior for the AC `BrowserToolbar`.